src/Common/Strings/StringUtil.ts

```diff
@@ -1,9 +1,6 @@
 ﻿export class StringUtil {
-  public static StringContainsSeparatedWord(str: string, word: string): boolean {
-    if (str === word ||
-      str.search(" " + word) !== -1 ||
-      str.search(word + " ") !== -1 ||
-      str.search(word + ".") !== -1) {
+  public static StringContainsSeparatedWord(str: string, wordRegExString: string): boolean {
+    if (new RegExp("( |^)" + wordRegExString + "([ .]|$)").test(str)) {
```

Comment:
it has been done: one RegEx to rule them all.

Reply:
Nice! But what is that | at the end of the line? Match jadajadajada or EOL?

Reply:
You mean `([ .]|$)`? That's white space, dot, or end of line

Reply:
Aah now I see it. Good